### ISSUE TITLE
[Feature Request] Make strikes persistent [PR welcome] 

### ISSUE THREAD

**Issue by ToasterUwU** [ManiMatter/decluttarr#219]
I looked for a way to do this, couldnt find anything. So im pretty sure this isnt a thing yet.

I would like to see a way to make the progress of the strikes persistent across restarts and even remaking of the container.

Im trying to download very niche things sometimes, so i set my strikes to an amount that last 3 days. I cant rely on their being loots of torrents out there, so i have to rely on giving the few that do exist all the time they might need.

Issue is that if i restart my server within 3 days, or an update for the containers gets applied, or i change my setup slightly and have to restart all containers because of it, the 288 max strikes get reset. At this rate it will never reach the limit and never actually do anything.

Prefered solution would be something simple like a file saved somewhere in the container so that making a bind volume to that location makes the progress data persist across even removing and remaking the container.

**Comment by ToasterUwU** [comment]
Im also happy to do this myself via PR if that would be welcome. I dont have time rn, but i should find some if i have to before new years.

**Comment by ManiMatter** [comment]
Hi 

this feature has been asked a couple of times but I dont have time either.

my initial thought is that the classes that track the strike write to a file. And when the script/container restarts, the init reads the file to get the history. We add a setting that can turn on/off this tracking between „reboots“

if you want to have a go, please feel free. Not planning of adding it myself

**Comment by ToasterUwU** [comment]
Yeah I was thinking the same thing basically. A json file of torrent hash key and strike integer would be more than good enough. 

I will take a look when I get the time and probably implement it.

**Comment by ManiMatter** [comment]
Closing as not planned from my side. Feel free to create a PR and happy to review
